### DISCUSSION TITLE
Laser made and modvfd

### DISCUSSION THREAD

**Discussion by yoft** [grblHAL/core#393] [Q&A] [answered]
Hi @terjeio ,
I have a diode laser wired into the PWM output of my board, and a VFD connected over modbus.
I'm looking into changing between the two easily. I've tired changing $32 in the middle of a tool change macro, which it gets upset about because the state is not IDLE.
After a bit more research, if I set the spindle cap.laser=0, and $32=1, will this cause the spindle to not utilise the laser mode features, but allow the PWM output to use laser mode?

Sorry if I've written that a bit weirdly...

Thanks!

**Comment by terjeio** [comment]
Spindle switching can be done by [tool number or M104](https://github.com/grblHAL/Plugins_spindle/#spindle-selection).

**Reply by yoft** [reply]
Sorry, I was referring to using PWM spindle in laser mode and Modbus spindle in normal mode.

**Reply by terjeio** [reply, marked as answer]
The $32 setting only affects laser mode capable PWM spindles, so just set it to 1. Mode switching will then be done automatically on a spindle change.

**Reply by yoft** [reply]
Brilliant! Thanks!
